ThirdPartyAdapters/applovin/applovin/src/main/java/com/applovin/mediation/AppLovinNativeAdapter.java

```diff
@@ -0,0 +1,54 @@
+package com.applovin.mediation;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.applovin.sdk.AppLovinSdk;
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.mediation.MediationNativeAdapter;
+import com.google.android.gms.ads.mediation.MediationNativeListener;
+import com.google.android.gms.ads.mediation.NativeMediationAdRequest;
+
+public class AppLovinNativeAdapter
+        implements MediationNativeAdapter {
+    protected static final String TAG = AppLovinNativeAdapter.class.getSimpleName();
+
+    public static final String KEY_EXTRA_AD_ID = "ad_id";
+    public static final String KEY_EXTRA_CAPTION_TEXT = "caption_text";
```

Comment:
Why are these keys in this class, when they are only used in another file?

Reply:
These keys will be used by the publisher to get the extras from the extras bundle.